sdk/tables/azure-data-tables/src/test/java/com/azure/data/tables/TableServiceAsyncClientTest.java

```diff
@@ -87,6 +86,51 @@ public void serviceCreateTable() {
             .verify();
     }
 
+
+    @Test
+    public void serviceAsyncClientCreateTableFailsWithIllegalArgumentExceptionIfTableNameInUri() {
+        // Arrange
+        String assertionString = "Table name found within client endpoint URI. Remove table name from endpoint.";
+        String tableName = testResourceNamer.randomName("test", 20);
+        final ClientSecretCredential credential = new ClientSecretCredentialBuilder()
+            .clientId(Configuration.getGlobalConfiguration().get("TABLES_CLIENT_ID", "clientId"))
+            .clientSecret(Configuration.getGlobalConfiguration().get("TABLES_CLIENT_SECRET", "clientSecret"))
+            .tenantId(testResourceNamer.randomUuid())
```

Comment:
The way we handle challenge-based authentication gives us the ability to figure out the actual tenant an account resides on even when being given a different one like we do here by providing a random value. That said, I think it should be better to provide the actual tenant of the account as we are not verifying that functionality in this particular test, we already do so in `serviceCreateTableWithMultipleTenants()`.

Reply:
The same applies for all the tests added in this PR.